### ISSUE TITLE
Improvements for v2

### ISSUE THREAD

**Issue by luciwoof** [RamBase/RamBaseApiDotNetSdk#6]
Draft of improvements for v2:
- Use System.Text.Json instead of Newtonsoft
- Fields that points to other json objects should point to corresponding C# objects after deserialization. (e.g. uncombinableWith should point to the corresponding NamedFilter)
- ???

More ideas will be added. Feel free to come with suggestions!

**Comment by andreny** [comment]
Support paging when getting from Api-resources. If nextpage exist continue to loop getting until all is received. Perhaps with a batchSize property/logic to make it possible to split in larger batches, which might be handy when processing the data.

**Comment by luciwoof** [comment]
I'm thinking that it might be a wrong approach to have access_token and such directly on the SDK. That ties an instance of the SDK to only one session, meaning that for each authorization grant we need a new instance of the SDK.

If we move this out of the SDK instance, and into its own authorizationGrant (or some other name) object will allow for an SDK instance to be used by several users. But the authorization object needs to be an argument into every method that calls the api.

Benefits:
- Authorization object can be cached (we might even provide methods for this?).
- Only one instance of the SDK.
- Only one HttpClient instance. Currently one is created for each SDK instance, which is not recommended! (https://docs.microsoft.com/en-gb/dotnet/api/system.net.http.httpclient?view=net-5.0) (Here we should look into the possibility to inject an instance)


Drawbacks:
- The authorization object needs to be used as an argument to every method.
- Harder to use for desktop application, or other applications where only one users is active at the same time.

The short version is that this makes sense for web applications with multiple simultaneous users, but less sense for desktop applications.

Do you have any thought on this @bpnilsen and @andreny ?

**Comment by bpnilsen** [comment]
I think all of these suggestions are good and we should consider all of them for the next version
